core/src/main/scala/org/apache/spark/deploy/DependencyUtils.scala

```diff
@@ -61,11 +62,12 @@ private[deploy] object DependencyUtils extends Logging {
       hadoopConf: Configuration,
       secMgr: SecurityManager): String = {
     val targetDir = Utils.createTempDir()
+    val fileSeparator = Pattern.quote(System.getProperty("file.separator"))
     Option(jars)
       .map {
         resolveGlobPaths(_, hadoopConf)
           .split(",")
-          .filterNot(_.contains(userJar.split("/").last))
+          .filterNot(_.contains(userJar.split(fileSeparator).last))
```

Comment:
Just split on `File.separator`, not a regex.

Reply:
Beyond the original purpose of this PR, is it better to move `userJar.split(fileSeparator).last` before line 66? This is because `userJar` is not changed in `map { ... }`.